src/Confluent.Kafka/AdminClient.cs

```diff
@@ -0,0 +1,374 @@
+
+            if (!manualPoll)
+            {
+                callbackCts = new CancellationTokenSource();
+                callbackTask = StartPollTask(callbackCts.Token);
+            }
+        }
+
+        /// <summary>
+        ///     Initializes a new Producer instance.
+        /// </summary>
+        /// <param name="config">
+        ///     librdkafka configuration parameters (refer to https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
+        ///     Topic configuration parameters are specified via the "default.topic.config" sub-dictionary config parameter.
+        /// </param>
+        public AdminClient(IEnumerable<KeyValuePair<string, object>> config)
+            : this(config, false) {}
+
+
+        /// <summary>
+        ///     Poll for callback events. You will not typically need
+        ///     to call this method. Only call on producer instances 
+        ///     where background polling is not enabled.
+        /// </summary>
+        /// <param name="millisecondsTimeout">
+        ///     The maximum period of time to block (in milliseconds) if no
+        ///     callback events are waiting or -1 to block indefinitely. 
+        ///     You should typically use a relatively short timout period 
+        ///     because this operation cannot be cancelled.
+        /// </param>
+        /// <returns>
+        ///     Returns the number of events served.
+        /// </returns>
+        public int Poll(int millisecondsTimeout)
+        {
+            if (!manualPoll)
+            {
+                throw new InvalidOperationException("Poll method called when manualPoll not enabled.");
+            }
+            return kafkaHandle.Poll((IntPtr)millisecondsTimeout);
+        }
+
+        /// <summary>
+        ///     Poll for callback events. You will not typically need
+        ///     to call this method. Only call on producer instances 
+        ///     where background polling is not enabled.
+        /// </summary>
+        /// <param name="timeout">
+        ///     The maximum period of time to block if no callback events
+        ///     are waiting. You should typically use a relatively short 
+        ///     timout period because this operation cannot be cancelled.
+        /// </param>
+        /// <returns>
+        ///     Returns the number of events served.
+        /// </returns>
+        public int Poll(TimeSpan timeout)
+            => Poll(timeout.TotalMillisecondsAsInt());
+        
+        /// <summary>
+        ///     Raised on critical errors, e.g. connection failures or all 
+        ///     brokers down. Note that the client will try to automatically 
+        ///     recover from errors - these errors should be seen as 
+        ///     informational rather than catastrophic
+        /// </summary>
+        /// <remarks>
+        ///     Called on the Producer poll thread.
+        /// </remarks>
+        public event EventHandler<Error> OnError;
```

Comment:
Ahh yes, I guess if we have another client type we're going to need all this stuff again. Perhaps we should have a base class `KafkaClient`. Maybe it should be`internal`.

Reply:
Yep. I tried to add one once, but it's not straight forward. Will be a nice refacto though :)